tests/test_features.py

```diff
@@ -10,14 +15,6 @@
 except KeyError:
     pass
 
-import matplotlib
```

Comment:
Depends on what you understand as "normalization along `axis=2`". We understood it as

 1. Iterate over selected axis
 2. Get elements in all other dimensions (e.g. column of a matrix or (N-1)D slice of an ND tensor)
 3. Normalize all these values

All values are touched exactly once.

The code below implements that in `normalize_axis0` and compares it to your
solution:

```python
import numpy


def normalize_axis0(S):
    # S is of dimension n

    # Iterate over axis 0
    for i in range(S.shape[0]):

        # S[i, ...] is of dimension n-1
        S[i, ...] /= numpy.max(S[i, ...])

    return S

A = numpy.ones((3, 3)) * 3.0
A[0, 0] = 12.0

print A / numpy.max(A, axis=0, keepdims=True)
print "-----"
print normalize_axis0(A)

print "----------"

A = numpy.ones((3, 3, 3)) * 3.0
A[0, 0, 0] = 12.0

print A / numpy.max(A, axis=0, keepdims=True)
print "-----"
print normalize_axis0(A)
```

In the 2D case those are interchangeable if you disregard the transposition error:

```
[[ 1.    1.    1.  ]
 [ 0.25  1.    1.  ]
 [ 0.25  1.    1.  ]]
-----
[[ 1.    0.25  0.25]
 [ 1.    1.    1.  ]
 [ 1.    1.    1.  ]]
```

In the 3D case however they differ.

Your code only regards a 1D slice (one "french fry") in the tensor, our implementation changes
a 2D slice (a whole "slice of bread").

```
[[[ 1.    1.    1.  ]
  [ 1.    1.    1.  ]
  [ 1.    1.    1.  ]]

 [[ 0.25  1.    1.  ]
  [ 1.    1.    1.  ]
  [ 1.    1.    1.  ]]

 [[ 0.25  1.    1.  ]
  [ 1.    1.    1.  ]
  [ 1.    1.    1.  ]]]
-----
[[[ 1.    0.25  0.25]
  [ 0.25  0.25  0.25]
  [ 0.25  0.25  0.25]]

 [[ 1.    1.    1.  ]
  [ 1.    1.    1.  ]
  [ 1.    1.    1.  ]]

 [[ 1.    1.    1.  ]
  [ 1.    1.    1.  ]
  [ 1.    1.    1.  ]]]
```